ncm-network/src/main/perl/network.pm

```diff
@@ -1,1183 +1,1728 @@
+    };
+
+    # VLAN support
+    &$makeline('vlan', bool => 'yesno');
 
+    push(@text, "ISALIAS=no") if ($iface->{vlan});
 
-        # VLAN support
-        # you do not need to set this for the VLAN device
-        $net{$iface}{vlan} = "true" if ($iface =~ m/^vlan\d+/);
-        if (exists($net{$iface}{vlan})) {
-            if ($net{$iface}{vlan} eq "true") {
-                $text .= "VLAN=yes\n";
-                # is this really needed?
-                $text .= "ISALIAS=no\n";
+    &$makeline('physdev');
+
+    my $vxlan = $iface->{plugin}->{vxlan};
+    if ($vxlan) {
+        if (!defined($vxlan->{vni})) {
+            my $device = $iface->{device} || $ifacename;
+            if ($device =~ /^vxlan([0-9]+)/) {
+                $vxlan->{vni} = $1;
             } else {
-                $text .= "VLAN=no\n";
+                $self->error("No VNI for vxlan plugin from devicename $device for $ifacename");
             }
+        };
+
+        push(@text, "VXLAN=yes");
+
+        my $vxlanml = _make_make_ifcfg_line($vxlan, \@text);
+        &$vxlanml('vni');
+
+        &$vxlanml('group', var => 'group_ipaddr');
+        # set remote and local ip addr twice
+        &$vxlanml('remote', var => 'remote_ipaddr');
+        &$vxlanml('remote', var => 'peer_outer_ipaddr');
+
+        &$vxlanml('local', var => 'local_ipaddr');
+        &$vxlanml('local', var => 'my_outer_ipaddr');
+
+        &$vxlanml('dstport');
+        &$vxlanml('gbp', bool => 'yesno');
+    };
+
+    return \@text;
+}
+
+# Return legacy ifcfg IPv4 route content
+sub make_ifcfg_route4_legacy
+{
+    my ($self, $routes) = @_;
+
+    my @text;
+    foreach my $idx (0 .. scalar(@$routes) -1) {
+        my $makeline = _make_make_ifcfg_line($routes->[$idx], \@text);
+        foreach my $attr (qw(address gateway netmask)) {
+            &$makeline($attr, var => "$attr$idx", def => ($attr eq 'netmask') ? '255.255.255.255' : undef);
         }
-        # interfaces named vlan need the physdev set and pointing to an existing interface
-        if ($iface =~ m/^vlan\d+/) {
-            if (exists($net{$iface}{physdev})) {
-                $text .= "PHYSDEV=".$net{$iface}{physdev}."\n";
+    }
+
+    return \@text;
+}
+
+# Return ifcfg route content in ip format
+sub make_ifcfg_ip_route
+{
+    my ($self, $flavour, $routes) = @_;
+
+    my @text;
+    foreach my $route (@$routes) {
+        if (!$route->{command}) {
+            my $ip;
+            if ($route->{prefix}) {
+                $ip = NetAddr::IP->new("$route->{address}/$route->{prefix}");
             } else {
-                $self->error("vlan device with vlan[0-9]{0-9} naming convention need physdev set.");
+                # in absence of netmask, NetAddr::IP uses 32 or 128
+                $ip = NetAddr::IP->new($route->{address}, $route->{netmask});
             }
+            # Generate it
+            $route->{command} = "$ip";
+            $route->{command} .= " via $route->{gateway}" if $route->{gateway};
```

Comment:
If the route doesn't have a gateway it should interpreted as a direct (link scoped) route and be followed by the device name, e.g.
```
10.16.184.0/22 dev eth0
```
or optionally:
```
10.16.184.0/22 dev eth0 scope link
```

i.e. it still has to be valid against the `ip route add` command schema

This preserves the previous behaviour and we rely on this working!

Reply:
See also: https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/6/html/Deployment_Guide/sec-Configuring_Static_Routes_in_ifcfg_files.html#sec-networkscripts-static-routes-ip-command

Reply:
the docs have `This is only required if the gateway is not set via DHCP and is not set globally in the /etc/sysconfig/network file` 
so you have no default gateway? or it is not via the device?

Reply:
This is not for setting default routes or "normal" routes via a gateway, we use this to allow access to multiple directly connected overlaid subnets, many of which the host itself may not have an IP for. In this case the host is essentially it's own gateway (aka. the route is link scoped).
The current component handles this correctly in the legacy route file format.

Reply:
ok, this looks like a trivial bug. will fix in a few minutes